core/module.go

```diff
@@ -233,7 +240,11 @@ func (mod *Module) FromRef(
 			return nil, fmt.Errorf("invalid local module ref is local relative to nil parent %q", moduleRefStr)
 		}
 		sourceDir = parentSrcDir
-		configPath = modules.NormalizeConfigPath(path.Join("/", path.Dir(parentSrcSubpath), modRef.Path))
+		configPath = modules.NormalizeConfigPath(filepath.Join(filepath.Dir(parentSrcSubpath), modRef.Path))
+
+		if strings.HasPrefix(configPath+"/", "../") {
+			return nil, fmt.Errorf("local module path %q is not under root", modRef.Path)
+		}
```

Comment:
This is actually the main "change" of this PR.

It ensures that a dependency "../foo" can't be loaded if the root is ".". Previously, this would silently succeed, and maybe actually try and load "./foo" instead.

However, we should *explicitly* error out.